src/Authentication/Traits/HasHmacTokens.php

```diff
@@ -156,4 +161,46 @@ public function setHmacToken(?AccessToken $accessToken): self
 
         return $this;
     }
+
+    /**
+     * Checks if the provided Access Token has expired.
+     *
+     * @return bool Returns true if Access Token has expired, false if not
+     */
+    public function hasHmacTokenExpired(?AccessToken $accessToken): bool
+    {
+        return $accessToken->expires !== null && $accessToken->expires->isBefore(Time::now());
+    }
+
+    /**
+     * Sets an expiration for Hmac Key/Token by ID.
+     *
+     * @param int  $id        AccessToken ID
+     * @param Time $expiresAt Expiration date
+     *
+     * @return bool Returns true if expiration date is set or updated.
+     */
+    public function setHmacTokenExpirationById(int $id, Time $expiresAt): bool
```

Comment:
Same question as with `setAccessTokenExpirationById()` - should we allow `?Time`?